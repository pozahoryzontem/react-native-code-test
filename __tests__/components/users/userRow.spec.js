import React from 'react';
import { create } from 'react-test-renderer';
import { Image } from 'react-native';
import UserRow from '../../../src/components/users/userRow';

describe('UserRow', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = create(<UserRow avatar="https://via.placeholder.com/300.png/09f/fffC" firstName="testFirstName" lastName="testLastName" />);
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
  describe('Image', () => {
    it('should load local image when onErrorImageLoad method called', () => {
      const component = create(<UserRow avatar="https://via.placeholder.com/300.png/09f/fffC" firstName="testFirstName" lastName="testLastName" />);
      const instance = component.root;
      const image = instance.findByType(Image);
      image.props.onError();
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
