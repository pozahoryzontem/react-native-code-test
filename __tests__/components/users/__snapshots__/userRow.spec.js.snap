// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRow Image should load local image when onErrorImageLoad method called 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "marginBottom": 10,
      "marginTop": 10,
    }
  }
>
  <Image
    onError={[Function]}
    source={
      Object {
        "testUri": "../../../src/assets/images/defaultAvatar.jpg",
      }
    }
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 40,
        "height": 80,
        "marginRight": 20,
        "width": 80,
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "Cochin",
        "fontSize": 20,
      }
    }
  >
    testFirstName testLastName
  </Text>
</View>
------------HTML PREVIEW---------------
<View style="align-items:center;flex:1;flex-direction:row;margin-bottom:10px;margin-top:10px;">
  <Image style="background-color:white;border-radius:40px;height:80px;margin-right:20px;width:80px;"></Image>
  <Text style="font-family:Cochin;font-size:20px;">
    testFirstName testLastName
  </Text>
</View>
`;

exports[`UserRow Rendering should match to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "marginBottom": 10,
      "marginTop": 10,
    }
  }
>
  <Image
    onError={[Function]}
    source={
      Object {
        "uri": "https://via.placeholder.com/300.png/09f/fffC",
      }
    }
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 40,
        "height": 80,
        "marginRight": 20,
        "width": 80,
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "Cochin",
        "fontSize": 20,
      }
    }
  >
    testFirstName testLastName
  </Text>
</View>
------------HTML PREVIEW---------------
<View style="align-items:center;flex:1;flex-direction:row;margin-bottom:10px;margin-top:10px;">
  <Image style="background-color:white;border-radius:40px;height:80px;margin-right:20px;width:80px;"></Image>
  <Text style="font-family:Cochin;font-size:20px;">
    testFirstName testLastName
  </Text>
</View>
`;
